core/client/fs/src/main/java/alluxio/client/block/RetryHandlingBlockMasterClient.java

```diff
@@ -107,6 +113,27 @@ public BlockInfo call() throws TException {
     });
   }
 
+  /**
+   * @param fields optional list of fields to query; if null all fields will be queried
+   * @return the {@link BlockMasterInfo} block master information
+   */
+  public synchronized BlockMasterInfo getBlockMasterInfo(final Set<BlockMasterInfoField> fields)
+      throws IOException {
+    return retryRPC(() -> {
+      Set<alluxio.thrift.BlockMasterInfoField> thriftFields = new HashSet<>();
```

Comment:
nit: to simplify the code and a minor optimization (avoid creating unused `HashSet` instance):
```java
Set<alluxio.thrift.BlockMasterInfoField> thriftFields = null;
if (fields != null) {
   thriftFields = new HashSet<>();
   for (BlockMasterInfoField field : fields) { ... }
}

```

Reply:
It looks much better and cleaner. Thanks for the suggestion!

Reply:
Done